### PR TITLE
Add table oci_core_instance_configuration closes #456

### LINKED ISSUE #456: Add table oci_core_instance_configuration
**References**
Terraform link : https://registry.terraform.io/providers/oracle/oci/latest/docs/resources/core_instance_configuration

GOSDK link :  https://pkg.go.dev/github.com/oracle/oci-go-sdk/v44/core#GetInstanceConfigurationRequest

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  display_name,
  id,
  time_created,
  region
from
  oci_core_instance_configuration
where
  time_created >= now() - interval '30' day;
+-------------------------------+----------------------------------------------------------------------------------------------------------+---------------------------+-------------+
| display_name                  | id                                                                                                       | time_created              | region      |
+-------------------------------+----------------------------------------------------------------------------------------------------------+---------------------------+-------------+
| instance-config-20230112-1015 | ocid1.instanceconfiguration.oc1.ap-mumbai-1.aaaaaaaauqtck6wy5radyrqq5ur2ny5izyjv6cl5wxablshgcgllwngzorna | 2023-01-12T10:15:45+05:30 | ap-mumbai-1 |
+-------------------------------+----------------------------------------------------------------------------------------------------------+---------------------------+-------------+

Time: 2.7s. Rows fetched: 1. Hydrate calls: 0.
> select
  display_name,
  id,
  time_created,
  region
from
  oci_core_instance_configuration;
+-------------------------------+----------------------------------------------------------------------------------------------------------+---------------------------+-------------+
| display_name                  | id                                                                                                       | time_created              | region      |
+-------------------------------+----------------------------------------------------------------------------------------------------------+---------------------------+-------------+
| instance-config-20230112-1015 | ocid1.instanceconfiguration.oc1.ap-mumbai-1.aaaaaaaauqtck6wy5radyrqq5ur2ny5izyjv6cl5wxablshgcgllwngzorna | 2023-01-12T10:15:45+05:30 | ap-mumbai-1 |
+-------------------------------+----------------------------------------------------------------------------------------------------------+---------------------------+-------------+

Time: 14ms. Rows fetched: 1 (cached). Hydrate calls: 0.
> 
```
</details>

### CODE REVIEW

**Comment by ParthaI** [comment]
Revisit the PR for integration test

File: docs/tables/oci_core_instance_configuration.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  comp.name as compartment_name
+from
+  oci_core_instance_configuration inst
+  inner join
+    oci_identity_compartment comp
+    on (inst.compartment_id = comp.id)
+order by
+  comp.name,
+  inst.region,
+  inst.shape;
```
Comment (karanpopat) [inline docs/tables/oci_core_instance_configuration.md:35]: Getting `Error: column inst.shape does not exist (SQLSTATE 42703)`

File: oci/table_oci_core_instance_configuration.go [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+	equalQuals := d.KeyColumnQuals
+
+	// Return nil, if given compartment_id doesn't match
+	if equalQuals["compartment_id"] != nil && compartment != equalQuals["compartment_id"].GetStringValue() {
+		return nil, nil
+	}
+
+	// Build request parameters
+	// request := buildCoreInstanceFilters(equalQuals)
```
Comment (karanpopat) [inline oci/table_oci_core_instance_configuration.go:132]: please remove commented code if not required anymore

File: docs/tables/oci_core_instance_configuration.md (hunk 1 above)
Comment (ParthaI) [reply docs/tables/oci_core_instance_configuration.md]: Resolved the error

File: oci/table_oci_core_instance_configuration.go (hunk 2 above)
Comment (ParthaI) [reply oci/table_oci_core_instance_configuration.go]: Removed the unnecessary commented codes.

File: oci/table_oci_core_instance_configuration.go [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+	case core.InstanceConfiguration:
+		id = *d.HydrateItem.(core.InstanceConfiguration).Id
+	case core.InstanceConfigurationSummary:
+		id = *d.HydrateItem.(core.InstanceConfigurationSummary).Id
+	}
+
+	return strings.Split(id, ".")[3], nil
+}
+
+// Build additional filters
```
[the remark this answers is not in the document]
Comment (ParthaI) [reply oci/table_oci_core_instance_configuration.go]: Removed the unnecessary commented codes.

**Review by misraved — APPROVED** [review]

**Pull request merged**